### PR TITLE
Remove skipStateRequest attribute and switch to comm-less models

### PR DESCRIPTION
**Pull request by SylvainCorlay**
Besides, the default_implementation of `_create_comm` now simply returns a rejected promise, so that we can always expect a promise.

### CODE REVIEW

**Comment by SylvainCorlay** [comment]
@jdfreder this removes `skipTestRequest` - and mostly removes code actually.

File: examples/development/web/index.js [hunk 1]
```diff
@@ -15,19 +16,18 @@ document.addEventListener("DOMContentLoaded", function(event) {
...
             model_name: 'WidgetModel',
             widget_class: 'ipywidgets.' + widgetType,
-            skipStateRequest: true
-            
+            model_id: uuid()
         // Create a view for the model.
-        }).then(function(model) { 
+        }).then(function(model) {
             console.log(widgetType + ' model created', model);
-
```
Comment (jdfreder) [inline examples/development/web/index.js]: If you're using atom, make sure to disable the whitespace package

Comment (SylvainCorlay) [inline examples/development/web/index.js]: I am using `vi` and it was a voluntary change :)

File: ipywidgets/static/widgets/js/manager-base.js [hunk 2]
```diff
@@ -386,13 +386,13 @@ define([
     };
 
     ManagerBase.prototype._create_comm = function(comm_target_name, model_id, metadata) {
-        throw new Error("Manager._create_comm not implemented");
+        return Promise.reject("No backend.");
     };
 
     ManagerBase.prototype._get_comm_info = function() {
-        throw new Error("Manager._get_comm_info not implemented");
+        return Promise.resolve({});
```
Comment (jdfreder) [inline ipywidgets/static/widgets/js/manager-base.js]: Shouldn't this be a Promise.reject ?

File: ipywidgets/static/widgets/js/manager-base.js [hunk 3]
```diff
@@ -179,19 +178,21 @@ define([
...
-                // Requesting the state to populate default values.
-                if (options.skipStateRequest) {
-                    return model;
-                } else {                    
-                    return model.request_state();
-                }
+                return model.request_state();
             });
+        }, function() {
+            // Comme Promise Rejected.
```
Comment (jdfreder) [inline ipywidgets/static/widgets/js/manager-base.js]: s/Comme/Comm

File: ipywidgets/static/widgets/js/manager-base.js [hunk 4]
```diff
@@ -179,19 +178,21 @@ define([
...
+            if (!options_clone.model_id) {
+                options_clone.model_id = utils.uuid();
```
Comment (jdfreder) [inline ipywidgets/static/widgets/js/manager-base.js]: Curious, should this be done inside new_model ?

Comment (SylvainCorlay) [inline ipywidgets/static/widgets/js/manager-base.js]: Maybe. 
I have not needed it in `new_model` yet...

File: ipywidgets/static/widgets/js/manager-base.js (hunk 2 above)
Comment (SylvainCorlay) [inline ipywidgets/static/widgets/js/manager-base.js]: `ManagerBase.set_state` does everything in the resolve handler of the comm_info...

**Comment by SylvainCorlay** [comment]
Done.

File: ipywidgets/static/widgets/js/manager-base.js (hunk 2 above)
Comment (jdfreder) [inline ipywidgets/static/widgets/js/manager-base.js]: That's okay, usually when I suffix a class with "Base" I mean this *must* be inherited to work.

**Comment by SylvainCorlay** [comment]
Rebased.

File: ipywidgets/static/widgets/js/manager-base.js (hunk 2 above)
Comment (jdfreder) [inline ipywidgets/static/widgets/js/manager-base.js]: I think this needs to be `return Promise.reject("No backend.");` because `_get_comm_info` is a method that returns a list of the existing comms from the backend.

If your problem with this is that you feel `set_state` should work without `_get_comm_info` being implemented, then `set_state` in the base should be modified to work without, and the one in the notebook WidgetManager should be implement so that it does.

Comment (SylvainCorlay) [inline ipywidgets/static/widgets/js/manager-base.js]: Could we make this a different PR?

Comment (jdfreder) [inline ipywidgets/static/widgets/js/manager-base.js]: Sure, but I would either leave the error that was there before, or use return `Promise.reject("No backend.");` here.

Comment (SylvainCorlay) [inline ipywidgets/static/widgets/js/manager-base.js]: OK, will add this right away.

**Comment by SylvainCorlay** [comment]
@jdfreder done!

**Comment by jdfreder** [comment]
Thanks @SylvainCorlay !
